com.onesignal.unity.android/Runtime/AndroidNotificationsManager.cs

```diff
@@ -0,0 +1,151 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * 1. The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * 2. All copies of substantial portions of the Software may only be used in connection
+ * with services provided by OneSignal.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+using UnityEngine;
+using System.Threading.Tasks;
+using OneSignalSDKNew.Notifications.Models;
+using System.Collections.Generic;
+
+namespace OneSignalSDKNew.Notifications {
+    internal sealed class AndroidNotificationsManager : INotificationsManager {
+        private readonly AndroidJavaObject _notifications;
+        
+        public AndroidNotificationsManager(AndroidJavaClass sdkClass) {
+            _notifications = sdkClass.CallStatic<AndroidJavaObject>("getNotifications");
+        }
+
+        public event NotificationWillShowDelegate WillShow; //INotificationWillShowInForegroundHandler
+        public event NotificationClickedDelegate Clicked; //INotificationClickHandler
+        public event PermissionChangedDelegate PermissionChanged; //IPermissionChangedHandler
+
+        public bool Permission {
+            get => _notifications.Call<bool>("getPermission");
+        }
+
+        public async Task<bool> RequestPermissionAsync(bool fallbackToSettings) {
+            var continuation = new BoolContinuation();
+            _notifications.Call<AndroidJavaObject>("requestPermission", fallbackToSettings, continuation.Proxy);
+            return await continuation;
+        }
+
+        public async Task ClearAllNotificationsAsync() {
+            var continuation = new Continuation();
+            _notifications.Call<AndroidJavaObject>("clearAllNotifications", continuation.Proxy);
+            await continuation;
+        }
+
+        public void Initialize() {
+            _notifications.Call("addPermissionChangedHandler", new IPermissionChangedHandler(this));
+            _notifications.Call("setNotificationWillShowInForegroundHandler", new INotificationWillShowInForegroundHandler(this));
+            _notifications.Call("setNotificationClickHandler", new INotificationClickHandler(this));
+        }
+
+        private sealed class IPermissionChangedHandler : OneSignalAwaitableAndroidJavaProxy<bool> {
```

Comment:
suggest these classes not start with "I", which typically indicates it's an interface.  Maybe `UnityPermissionChangedHandler` or `InternalPermissionChangedHandler`, something like that.

Reply:
Oh I see, going to change it to "Internal" next